src/tracer-config.test.ts

```diff
@@ -0,0 +1,307 @@
+    process.env['SEMMLE_PRELOAD_libtrace'] = 'abc';
+    process.env['SEMMLE_RUNNER'] = 'abc';
+    process.env['SEMMLE_COPY_EXECUTABLES_ROOT'] = 'abc';
+    process.env['SEMMLE_DEPTRACE_SOCKET'] = 'abc';
+    process.env['SEMMLE_JAVA_TOOL_OPTIONS'] = 'abc';
+    process.env['SEMMLE_DEPTRACE_SOCKET'] = 'abc';
+    process.env['CODEQL_VAR'] = 'abc';
+
+    const result = await tracerConfig(codeQL, config, Language.javascript);
+    t.deepEqual(result, {
+      spec: 'abc',
+      env: {
+        'baz': 'qux',
+        'SEMMLE_PRELOAD_libtrace': 'SEMMLE_PRELOAD_libtrace',
+        'SEMMLE_RUNNER': 'SEMMLE_RUNNER',
+        'SEMMLE_COPY_EXECUTABLES_ROOT': 'SEMMLE_COPY_EXECUTABLES_ROOT',
+        'SEMMLE_DEPTRACE_SOCKET': 'SEMMLE_DEPTRACE_SOCKET',
+        'SEMMLE_JAVA_TOOL_OPTIONS': 'SEMMLE_JAVA_TOOL_OPTIONS',
+        'CODEQL_VAR': 'CODEQL_VAR',
+      }
+    });
+  });
+});
+
+test('concatTracerConfigs - minimal', async t => {
+  await util.withTmpDir(async tmpDir => {
+    const config = getTestConfig(tmpDir);
+
+    const spec1 = path.join(tmpDir, 'spec1');
+    fs.writeFileSync(spec1, 'foo.log\n2\nabc\ndef');
+    const tc1 = {
+      spec: spec1,
+      env: {
+        'a': 'a',
+        'b': 'b',
+      }
+    };
+
+    const spec2 = path.join(tmpDir, 'spec2');
+    fs.writeFileSync(spec2, 'foo.log\n1\nghi');
+    const tc2 = {
+      spec: spec2,
+      env: {
+        'c': 'c',
+      }
+    };
+
+    const result = concatTracerConfigs({ 'javascript': tc1, 'python': tc2 }, config);
+    t.deepEqual(result, {
+      spec: path.join(tmpDir, 'compound-spec'),
+      env: {
+        'a': 'a',
+        'b': 'b',
+        'c': 'c',
+      }
+    });
+    t.true(fs.existsSync(result.spec));
+    t.deepEqual(
+      fs.readFileSync(result.spec, 'utf8'),
+      path.join(tmpDir, 'compound-build-tracer.log') + '\n3\nabc\ndef\nghi');
+  });
+});
+
+test('concatTracerConfigs - conflicting env vars', async t => {
+  await util.withTmpDir(async tmpDir => {
+    const config = getTestConfig(tmpDir);
+
+    const spec = path.join(tmpDir, 'spec');
+    fs.writeFileSync(spec, 'foo.log\n0');
+
+    // Ok if env vars have the same name and the same value
+    t.deepEqual(
+      concatTracerConfigs(
+        {
+          'javascript': {spec: spec, env: {'a': 'a', 'b': 'b'}},
+          'python': {spec: spec, env: {'b': 'b', 'c': 'c'}},
+        },
+        config).env,
+      {
+        'a': 'a',
+        'b': 'b',
+        'c': 'c',
+      });
+
+    // Throws if env vars have same name but different values
+    const e = t.throws(() =>
+      concatTracerConfigs(
+        {
+          'javascript': {spec: spec, env: {'a': 'a', 'b': 'b'}},
+          'python': {spec: spec, env: {'b': 'c'}},
+        },
+        config));
+    t.deepEqual(e.message, 'Incompatible values in environment parameter b: b and c');
+  });
+});
+
+// If cpp is present then it's spec lines always come at the end
+test('concatTracerConfigs - cpp comes last', async t => {
```

Comment:
```suggestion
test('concatTracerConfigs - C++ spec lines come last if present', async t => {
```

If you do keep the comment, remove the apostrophe from `it's` 😉